src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java

```diff
@@ -63,28 +80,46 @@ public String getConfirmation(@PathVariable String companyNumber,
             return ERROR_VIEW;
         }
 
-
         // If the payment is anything but paid return user to beginning of journey
+        PayableLateFilingPenalty payableLateFilingPenalty;
+        CompanyProfileApi companyProfileApi;
+        try {
+            companyProfileApi = companyService.getCompanyProfile(companyNumber);
+            payableLateFilingPenalty = payablePenaltyService
+                    .getPayableLateFilingPenalty(companyNumber, penaltyId);
+        } catch (ServiceException ex) {
+            LOGGER.errorRequest(request, ex.getMessage(), ex);
+            return ERROR_VIEW;
+        }
+
         if (!paymentStatus.equals("paid")) {
             LOGGER.info("Payment status is " + paymentStatus + " and not of status 'paid', returning to beginning of journey");
-            PayableLateFilingPenalty payableLateFilingPenalty;
-            try {
-                payableLateFilingPenalty = payablePenaltyService
-                        .getPayableLateFilingPenalty(companyNumber, penaltyId);
-            } catch (ServiceException ex) {
-                LOGGER.errorRequest(request, ex.getMessage(), ex);
-                return ERROR_VIEW;
-            }
-
             Map<String, String> links = payableLateFilingPenalty.getLinks();
             return UrlBasedViewResolver.REDIRECT_URL_PREFIX + links.get("resume_journey_uri");
```

Comment:
Do we need the 'links' map declaration? I would go for:
`return UrlBasedViewResolver.REDIRECT_URL_PREFIX + payableLateFilingPenalty.getLinks().get("resume_journey_uri");`
line 99 -> Remove the unused line

Reply:
Update as suggested